app/assets/javascripts/common/drug_stocks.js

```diff
@@ -0,0 +1,21 @@
+// Allow users to enter only some of the protocol drugs in the list, while validating whatever is entered.
+// i.e, if stock value is not blank, then in_stock is a required field.
+function markRequiredFields(facility_id) {
+    let protocol_drugs = document.querySelectorAll('#drug_report_form_'+ facility_id + ' .protocol-drug-stock-inputs');
+    protocol_drugs.forEach(function(protocol_drug) {
+        let received = protocol_drug.querySelector('.received');
+        let in_stock = protocol_drug.querySelector('.in_stock');
+        if(received.value.trim() != '') {
+            in_stock.required = true;
+        } else {
+            in_stock.required = false;
+        }
+    });
+};
+
+function validateAndCloseDrugStockForm(facility_id) {
```

Comment:
Non-blocking suggestion:

Might be nice to class-ify this JS if possible. We've been trying to move to a little bit of prototypical JS lately. It doesn't entirely apply in this scenario because the function acts as a closure that doesn't arbitrarily get called, but namespacing will ensure that for sure.

It could be something like `new DrugStock().markRequired()` or something.

[ref](https://github.com/simpledotorg/simple-server/blob/51be010af90cd5de81da56563456026ebaa52ff3/app/assets/javascripts/common/admins.js#L200), [ref](https://github.com/simpledotorg/simple-server/blob/51be010af90cd5de81da56563456026ebaa52ff3/app/assets/javascripts/common/user_teleconsult_search.js).

Reply:
This js is gone now